### ISSUE TITLE
Closing to many channels simultaneously can kill the node

### ISSUE THREAD

**Issue by DerEwige** [ACINQ/eclair#2702]
My `local.htlc_infos` table has grown to over 100 million entries.
The average channel has around 2 million entries in the table.
Deleting one of the channels from the table can take up to 20 minutes.

This is usually not a problem.
The delete job will just run in the DB and during this time the node reacts just a bit slower.

Yesterday I decided to clean out some channels and close 6 channels in short succession.
All transaction from the 6 closes got mined within 2 minutes and shortly after this my node entered a restart circle.

This error would happen:
`2023-06-26 22:35:59,593 ERROR- eclair is shutting down because of a fatal database errorjava.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30002ms.
`
Node shuts down, restarts, runs for 2 minutes then shuts down again...
This would happen over and over.

The problem was this:

I had `eclair.db.postgres.pool.size`=4
When the node started up it would take a moment to realize that the 6 channel close TX were already mined and started the delete
Those deletes would run for several minutes and block all 4 connections to the db.
This would then lead to the shutdown of the node.
Because the node was shut down the delete transactions would not get committed and the circle would start over on next tart of node.

Increasing eclair.db.postgres.pool.size to 8 allowed the node to use 6 connections for the deletes while 2 connections would enough to handle the rest of the DB load.

How to reproduce:
	- Grow your `local.htlc_infos` to the point where the delete of one channel takes more than 5 minutes.
	- Close double the number of channels than your `eclair.db.postgres.pool.size`
	- Shut down your node and wait for all closing TX to be mined (this is to make sure the deletes all happen at once)
	- Start your node (node will then fill up the DB connections with deletes and run in a DB timeout)

Version:
version=0.9.0 commit=623f7e4 with postgres DB

**Comment by pm47** [comment]
Try increasing `eclair.db.postgres.pool.connection-timeout`, it defaults to 30s.

For the record, attempting 100 million payments that probably don't fulfill is not only non-sensical, it is a class of dos attack on the network that protocol developers including us actively work towards preventing.

**Comment by rorp** [comment]
> For the record, attempting 100 million payments that probably don't fulfill is not only non-sensical, it is a class of dos attack on the network that protocol developers including us actively work towards preventing.

The number doesn't seem that non-sensical if you rebalance channels a lot.

**Comment by DerEwige** [comment]
Yes, I rebalance a lot and that is how my node reaches this numbers.
But a big zero fee routing node would have similar numbers.

I believe a simple fix could be to make the function removeChannel() synchronized, so only one channel gets removed at a time.
There might be other SQL statements, that run a long time, but so far that is the only one that gave me problems.
